rosidl_generator_cpp/resource/msg_Struct.h.template

```diff
@@ -106,6 +106,26 @@ for field in spec.fields:
   typedef std::shared_ptr< @(cpp_full_name)<ContainerAllocator> > Ptr;
   typedef std::shared_ptr< @(cpp_full_name)<ContainerAllocator> const> ConstPtr;
 
+@# generate comparion operators
+  bool operator==(const @(spec.base_type.type)_ & other) const
+  {
+@[for field in spec.fields]@
+    if (this->@(field.name) != other.@(field.name)) {
+      return false;
+    }
+@[end for]@
+    return true;
+  }
+  bool operator!=(const @(spec.base_type.type)_ & other) const
```

Comment:
Good point to get rid of the duplicate logic. I updated it to use the following since we don't want lhs conversion here:

    return !this->operator==(other);